File: UPPE/UPPE_Output.m
N_row_ = 621;
N_col_ = 20;

fileID = fopen('output/A_w_R.bin');
A_w_R = fread(fileID,[N_row_ N_col_],'double');

figure
for i = 1:20
    subplot(4, 5, i)
    plot(A_w_R(:, i))
    title('A\_w\_R[i]')
end
figure
plot(A_w_R)
title('A\_w\_R Combined')

fileID = fopen('output/A_w_I.bin');
A_w_I = fread(fileID,[N_row_ N_col_],'double');

figure
for i = 1:20
    subplot(4, 5, i)
    plot(A_w_I(:, i))
    title('A\_w\_I[i]')
end
figure
plot(A_w_I)
title('A\_w\_I Combined')

fileID = fopen('output/w_active.bin');
w_active = fread(fileID,[1, N_col_],'double');
figure
plot(w_active)
title('w\_active')

A_w = complex(A_w_R, A_w_I);
A_w_abs_2 = (abs(A_w)).^2;

figure
for i = 1:20
    subplot(4, 5, i)
    plot(A_w_abs_2(:, i))
    title('A\_w\_abs\_2[i]')
end
figure
plot(A_w_abs_2)
title('A\_w\_abs\_2 Combined')

A_t = fft(A_w);
A_t_abs_2 = (abs(A_t)).^2;

figure
for i = 1:20
    subplot(4, 5, i)
    plot(A_t_abs_2(:, i))
    title('A\_t\_abs\_2[i]')
end
figure
plot(A_t_abs_2)
title('A\_t\_abs\_2 Combined')

figure
plot(w_active_t_abs_2)
title('w\_active\_t\_abs\_2')

A_t_abs_2_sum = sum(A_t_abs_2, 2);
figure
plot(A_t_abs_2_sum)

figure
plot(A_t_abs_2_sum); hold on;
plot(A_t_abs_2)

File: UPPE/UPPE_Output.py
import struct
import numpy as np
from matplotlib import pyplot as plt

# !/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
Created on Wed Feb  1 14:50:42 2017

@author: sms1n16
"""

N_row_ = 621
N_col_ = 20

fileName = "./output/A_w_R.bin"
f = open(fileName, mode='rb')
content = f.read(8*N_row_*N_col_)

data = struct.unpack("d"*(len(content) // (8)), content)
# N = np.fromfile(fileName, dtype=np.float64)

A_w_R = np.zeros([N_col_, N_row_])
for i in range(N_col_):
    A_w_R[i, :] = data[i*N_row_:i*N_row_+N_row_]

plt.figure()
plt.plot(A_w_R.T)
plt.show()


fileName = "./output/A_w_I.bin"
f = open(fileName, mode='rb')
content = f.read(8*N_row_*N_col_)

data = struct.unpack("d"*(len(content) // (8)), content)
# N = np.fromfile(fileName, dtype=np.float64)

A_w_I = np.zeros([N_col_, N_row_])
for i in range(N_col_):
    A_w_I[i, :] = data[i*N_row_:i*N_row_+N_row_]

plt.figure()
plt.plot(A_w_I.T)
plt.show()


fileName = "./output/w_active.bin"
f = open(fileName, mode='rb')
content = f.read(8*N_row_*1)

data = struct.unpack("d"*(len(content) // (8)), content)
# N = np.fromfile(fileName, dtype=np.float64)

w_active = np.zeros([1, N_row_])
w_active[0, :] = data[0:N_row_]

plt.figure()
plt.plot(w_active[0])
plt.show()
